### PR TITLE
[SPARK-20555][SQL] Fix mapping of Oracle DECIMAL types to Spark types

### PR DESCRIPTION
**Pull request by gaborfeher**
## What changes were proposed in this pull request?

This change removes the custom code from `OracleDialect`, that was mapping some of Oracle's `DECIMAL(X, Y)` types to `BooleanType`, `IntegerType`, `LongType` or `FloatType`.

## How was this patch tested?

I used the following command to run the docker-based integration tests:
```
./build/mvn -Pdocker-integration-tests -pl :spark-docker-integration-tests_2.11 test
```

### CODE REVIEW

**Comment by gatorsmile** [comment]
ok to test

File: external/docker-integration-tests/src/test/scala/org/apache/spark/sql/jdbc/OracleIntegrationSuite.scala [hunk 1]
```diff
@@ -148,27 +172,28 @@ class OracleIntegrationSuite extends DockerJDBCIntegrationSuite with SharedSQLCo
...
     // verify the value is the inserted correct or not
     val values = rows(0)
-    assert(values.getBoolean(0).equals(booleanVal))
-    assert(values.getInt(1).equals(integerVal))
-    assert(values.getLong(2).equals(longVal))
-    assert(values.getFloat(3).equals(floatVal))
-    assert(values.getFloat(4).equals(doubleVal.toFloat))
-    assert(values.getInt(5).equals(byteVal.toInt))
-    assert(values.getInt(6).equals(shortVal.toInt))
+    assert(values.getDecimal(0).compareTo(BigDecimal.valueOf(booleanVal ? 1 : 0)) == 0)
```
Comment (gatorsmile) [inline external/docker-integration-tests/src/test/scala/org/apache/spark/sql/jdbc/OracleIntegrationSuite.scala]: This does not work. The conditional operator `? :` is not supported by Scala.

**Comment by gatorsmile** [comment]
ping @gaborfeher

**Comment by gaborfeher** [comment]
Hi, thanks for the review! I definitely want to continue, but I'll only have more time to look into this next week. (I don't exactly know how I ended up with that broken test, because I was able to run some tests locally. I think, at least.)

**Comment by gatorsmile** [comment]
This is a pretty serious bug. Let me resolve the failure as a separate PR. Will give all the credits to you when we merging the PR. Thank you for your fix!

**Comment by gaborfeher** [comment]
Thank you, I am happy to see the issue is taken care of! (I don't know the customs here, we can also share the credit.)

**Comment by gatorsmile** [comment]
@gaborfeher Since the PR has been merged, could you please close it? Thanks!

**Pull request closed without merging**